openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -568,6 +569,13 @@ def post(self, request):
 
         data = request.POST.copy()
         self._handle_terms_of_service(data)
+        try:
+            data = PreRegisterFilter.run(form_data=data)
```

Comment:
To the comment about removing the password.
I can see reasons to leave the password in, for example auditing the security by calling an API at have i been pwned, but that is a small possibility/feature at the cost of opening up a big hole in the security since other plugins could use the password for the wrong reasons.

In any case it would be easy to copy the data dict and remove anything in the list of censored strings (https://github.com/edx/edx-platform/blob/4f4be6538ae008a6d65cd7cc49d776aebfa9e40a/common/djangoapps/track/middleware.py#L64)

Reply:
I'm a little ambivalent on this. Anything running with this level of permission can already do anything they want, security wise. So this is to prevent people accidentally doing horribly unsecure things. That being said, I can think of other fields that edX itself would also consider extremely sensitive, and not want filters to operate on (like certain demographics information).

In any case, I think it's fine to block these off for now, and then we can always add them back in later. I'm imagining that the LMS would always pass through the entire form data, and the `PreRegisterFilter` implementation would decide how much to pass on to the individual filters or not? And then if we really want to get fancy down the road, we could make that configurable via filter-level configuration?